components/template/template-application-ui-angular/src/main/resources/META-INF/dirigible/template-application-ui-angular/ui/perspective/manage/controller.js.template

```diff
@@ -172,19 +172,47 @@ angular.module('page', ["ideUI", "ideView", "entityApi"])
 		$scope.options${property.name} = [];
 #end
 #end
-#foreach ($property in $properties)
-#if($property.widgetType == "DROPDOWN")
 
-		${dollar}http.get("/services/ts/${projectName}/gen/api/${property.relationshipEntityPerspectiveName}/${property.relationshipEntityName}Service.ts").then(function (response) {
-			${dollar}scope.options${property.name} = response.data.map(e => {
-				return {
-					value: e.${property.widgetDropDownKey},
-					text: e.${property.widgetDropDownValue}
-				}
-			});
-		});
+#foreach ($property in $properties)
+#if ($property.widgetType == "DROPDOWN")
+#if ($referencedProjections.length != 0)
+#set ($foundReferenceProjection = false)
+#foreach ($referencedProjection in $referencedProjections)
+                #if ($referencedProjection.name == $property.relationshipEntityName && !$foundReferenceProjection)
+         ${dollar}http.get("/services/ts/${referencedProjection.project}/gen/api/${property.relationshipEntityPerspectiveName}/${property.relationshipEntityName}Service.ts").then(function (response) {
+                 ${dollar}scope.options${property.name} = response.data.map(e => {
+                     return {
+                         value: e.${property.widgetDropDownKey},
+                         text: e.${property.widgetDropDownValue}
+                     }
+                 });
+             });
+#set ($foundReferenceProjection = true)
+#end
+#end
+#if (!$foundReferenceProjection)
+      ${dollar}http.get("/services/ts/${projectName}/gen/api/${property.relationshipEntityPerspectiveName}/${property.relationshipEntityName}Service.ts").then(function (response) {
+              ${dollar}scope.options${property.name} = response.data.map(e => {
+                  return {
+                      value: e.${property.widgetDropDownKey},
+                      text: e.${property.widgetDropDownValue}
+                  }
+              });
+          });
+#end
+#else
+      ${dollar}http.get("/services/ts/${projectName}/gen/api/${property.relationshipEntityPerspectiveName}/${property.relationshipEntityName}Service.ts").then(function (response) {
+              ${dollar}scope.options${property.name} = response.data.map(e => {
+                  return {
+                      value: e.${property.widgetDropDownKey},
+                      text: e.${property.widgetDropDownValue}
+                  }
+              });
+          });
 #end
 #end
+#end
```

Comment:
I think it would be a better for the whole logic to happen inside the `parameterUtils.js`, as here in the template it's too hard to understand it. In the end only the url is different:
```
"/services/ts/${projectName}/gen/api/${property.relationshipEntityPerspectiveName}/${property.relationshipEntityName}Service.ts"
```
vs
```
"/services/ts/${referencedProjection.project}/gen/api/${property.relationshipEntityPerspectiveName}/${property.relationshipEntityName}Service.ts"
```